plugins/TagFix_Vatin.py

```diff
@@ -0,0 +1,89 @@
+#-*- coding: utf-8 -*-
+
+###########################################################################
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+
+from plugins.Plugin import Plugin
+import re
+
+class TagFix_Vatin(Plugin):
+
+    # ref:vatin is a tag to add the VAT identification number.
+    # The usual syntax is <country-code><VAT-number>
+    # Examples:
+    # ref:vatin=IT12345678901 for a business in Italy
+    # ref:vatin=DE115055186 for a business in Germany
+
+    # XXX Enable more countries
+    only_for = ["IT"]
+
+    def init(self, logger):
+        Plugin.init(self, logger)
+        self.errors[99999] = {"item": 9999, "level": 3, "tag": ["ref"], "desc": T_(u'Invalid value of tag "ref:vatin"') }
+
+    # https://it.wikipedia.org/wiki/Partita_IVA
+    def it_vatin(self, vatin):
+        if (len(vatin) != 11):
+            return False
+
+        # A Luhn algorithm implementation
+        x = 0
+        y = 0
+        z = 0
+        for i in range(0, 10, 2):
+            x += int(vatin[i:i + 1])
+        for i in range(1, 10, 2):
+            y += 2 * (int(vatin[i:i + 1]))
+            if int(vatin[i:i + 1]) >= 5:
+                z = z + 1
+        t = (x + y + z) % 10
+        c = int(vatin[10:11])
+        return c == (10 - t) % 10
+
+    def node(self, data, tags):
+        if "ref:vatin" in tags:
+            x = re.split("^([A-Z]{2})(.+)$", tags["ref:vatin"])
```

Comment:
Precompile the regex in the init. But here you does not need a regex `if tags["ref:vatin"].startswith('IT')`

Reply:
and `tags["ref:vatin"]|2:]` for the last part.

Reply:
Well, consider that any country code (IT, DE, FR,...) followed by a code (alphanum) is valid...
I am not a python expert, how should I pre-compile it?

Reply:
Not sure what you want to achieve here. But avoid using regex if not required.
You can check the first two are country code like https://docs.python.org/fr/2/library/stdtypes.html?highlight=isalpha#str.isalpha

To compile regex https://docs.python.org/2/library/re.html#re.compile

Reply:
The idea was to check that the pattern <CountryCode><VAT> is respected, but could achieved in a later revision of this file